phoenix-core/src/it/java/org/apache/phoenix/end2end/UpsertValuesIT.java

```diff
@@ -480,10 +488,14 @@ public void testBatchedUpsert() throws Exception {
             pstmt.addBatch();
             pstmt.executeBatch();
             conn.commit();
+            PhoenixConnection pConn = conn.unwrap(PhoenixConnection.class);
+            Map<String, Map<MetricType, Long>> mutationMetrics = pConn.getMutationMetrics();
```

Comment:
You should also check that the underlying mutation state inside connection has joined all the statements in the batch. You can check the value of `connection.getMutationState().getUpdateCount()`

Reply:
Actually, `updateCount` in `MutationState` gets reset when `connection.commit()` is called and since it gets called implicitly as part of `Statement.executeBatch()`, it will always be `0` if we check from this test. Do you have any idea on the appropriate location?